bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/graphics/SVGUtil.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse
+ * Public License 2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors: Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.swt.graphics;
+
+import java.io.*;
+import java.nio.charset.*;
+
+/**
+ * Utility class for handling SVG-related operations.
+ *
+ * @since 3.129
+ */
+public class SVGUtil {
+
+	/**
+	 * Determines whether the given {@link InputStream} contains a SVG file.
+	 *
+	 * @param data byte array to check.
+	 * @return {@code true} if the input stream contains SVG content; {@code false}
+	 *         otherwise.
+	 * @throws IOException              if an error occurs while reading the stream.
+	 * @throws IllegalArgumentException if the input stream is {@code null}.
+	 */
+	public static boolean isSVGFile(byte[] data) throws IOException {
+		String content = new String(data, 0, Math.min(data.length, 512), StandardCharsets.UTF_8);
+		return content.contains("<svg");
```

Comment:
This is wrong anyways, svg can have xml document declaration like this

```
<?xml version="1.0" encoding="UTF-8"?>
<svg xmlns="http://www.w3.org/2000/svg" ...>
```

For the sake of this check, simply check if the first byte is an `<` (what is invalid for any other image format to start with we currently support).

Also I think the rasterizer should not need to check this again, the caller is responsible to only pass valid data or will get an error.

Reply:
Thank you for the remark, I will fix it. But can you elaborate why this shouldn't work with the snippet you provided? The `<svg` tag in the snippet would be read by my code.

Reply:
This is changed in 3bc362193af1f1918abbf985346dee5f39de997d. The caller is now responsible for making sure the InputStream contains an SVGFile. Also only the first byte is checked.